drivers/pwm/pwm_nrfx.c

```diff
@@ -5,93 +5,236 @@
  */
 #include <nrfx_pwm.h>
 #include <pwm.h>
+#include <hal/nrf_gpio.h>
+#include <stdbool.h>
 
 #define LOG_LEVEL CONFIG_PWM_LOG_LEVEL
 #include <logging/log.h>
 LOG_MODULE_REGISTER(pwm_nrfx);
 
-#define PWM_NRFX_CH_VALUE_NORMAL        (1UL << 15)
+#define PWM_NRFX_CH_POLARITY_MASK       BIT(15)
+#define PWM_NRFX_CH_PULSE_CYCLES_MASK   BIT_MASK(15)
+#define PWM_NRFX_CH_VALUE_NORMAL        PWM_NRFX_CH_POLARITY_MASK
 #define PWM_NRFX_CH_VALUE_INVERTED      (0)
+#define PWM_NRFX_CH_PIN_MASK            ~NRFX_PWM_PIN_INVERTED
 
 struct pwm_nrfx_config {
 	nrfx_pwm_t pwm;
-	nrfx_pwm_config_t config;
+	nrfx_pwm_config_t initial_config;
 	nrf_pwm_sequence_t seq;
 };
 
 struct pwm_nrfx_data {
+	u32_t period_cycles;
 	u16_t current[NRF_PWM_CHANNEL_COUNT];
-	u16_t top_value;
+	u16_t countertop;
+	u8_t  prescaler;
 };
 
+static u32_t pwm_period_check_and_set(const struct pwm_nrfx_config *pconfig,
+				      struct pwm_nrfx_data *pdata,
```

Comment:
If 'p' in pdata and pconfig stand for pointer it should be [removed.](https://www.kernel.org/doc/html/v4.10/process/coding-style.html#naming)

Reply:
Done. Removed all p prefix on variable names.